### PR TITLE
feat: add tests CI

### PR DESCRIPTION
**Pull request by McPatate**
Adding test CI

Wasted too much time because ClamAV doesn't support relative path for the db dir 😬  : https://github.com/Cisco-Talos/clamav/issues/407

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -685,3 +786,15 @@ mod tests {
         Ok(())
     }
 }
+
+#[cfg(doctest)]
+mod test_readme {
```
Comment (LevitatingOrange) [inline src/lib.rs:791]: Cool Idea! Does this do anything yet though? Otherwise maybe we comment this out and do that at a later date.

**Review by LevitatingOrange — CHANGES_REQUESTED** [review]
Thangs again for this PR looks great! I like the Idea of testing the readme also, but does that code do anything yet? Otherwise I would just comment it out for now. 

Looks really good, I'll merge once we decided on the above.

File: src/lib.rs (hunk 1 above)
Comment (McPatate) [reply src/lib.rs:791]: Yep it makes the readme code bits runnable, so when you run `cargo test` it checks the code compiles.

The downside is that if you put other "\`\`\`" blocks in the readme it tries to compile them even if it's not rust code. I'm not too familiar with macro syntax but surely there's a way to only check "\`\`\`rust" code bits.

Comment (LevitatingOrange) [reply src/lib.rs:791]: Ahh cool, as you can see unfortunately rust's macro syntax is a blind spot of me too haha. Lets just merge it as-is, we can worry about that at another point in time.

**Comment by LevitatingOrange** [comment]
Thanks again for your effort! :smiley:

**Pull request merged**